### ISSUE TITLE
Switch to ES6 style classes for PluginModule & derived classes

### ISSUE THREAD

**Issue by NuSkooler** [NuSkooler/enigma-bbs#61]
**Issue**:
Currently PluginModule and derived classes use older pre-ES6 style "classes". It would be nice & cleaner to move to ES6.

**Midterm**:
If it's possible to mix the two, at least newer classes could be ES6 but they would have to derive from the older style. I'm not sure if this is possible.

**Comment by tracker1** [comment]
Without using babel, there are some features you won't be able to use such as decorators, and static properties...  But the implementation would probably look a lot cleaner... IIRC node 4.x does have class syntax.  Also missing are destructuring object assignment and a handful of other niceties.

http://node.green/

**Comment by NuSkooler** [comment]
@tracker1 Thanks for the link, that's very handy! ENiGMA is already using ES6 classes in a lot of areas (ie https://github.com/NuSkooler/enigma-bbs/blob/master/core/ftn_address.js), but the original PluginModule and derived code was pre-classes support. What I'm not clear on is if I can somehow mix them (e.g. update PluginModule to ES6 class and not the derived or vice versa) or if it's all or nothing (which is obviously a much larger at once project)

There is another issue in the system for updating lodash... I think it would be good to move everything to Node 6.x LTS as a base requirement overall.

**Comment by tracker1** [comment]
ES6 classes are defined as sugar over prototypal inheritance... so you can definitely mix the two.

**Comment by tracker1** [comment]
Just curious, thinking on finally getting this setup over the weekend to work on... would you be willing to accept a PR if core and mods were moved into a _src directory, and a task to run them through babel into the core/mods directories as part of an npm run build, or have a prestart task that conditionally runs build?

I'd like to start on a webserver & webclient (webserver as part of core), using react+redux, but to use it effectively on the server-side, really want decorators and a few other bits (stage-1 or even stage-2 would help a lot),  I'd like the server to run with the rest of the BBS process, so state (online users, etc) can be shared.

```
./core => ./_src/core => ouput(./core)
./mods => ./_src/mods => ouput(./mods)
n/a => _src/web-server => ouput(./web-server) (references to/from core)
n/a => _src/web-client => output(./web-client)
```

**Comment by NuSkooler** [comment]
@tracker1 That doesn't really align with ENiGMA's goals. 

I have on my list to move the current two servers (telnet & SSH) to something such as `core/servers/login/` and name them login_servers. This would allow other servers such as WebSocket, NNTP, HTTP(S), and so on to be generic servers.

TL;DR: With the above, one could implement a webserver (using whatever you like). What you would not get out of this is babel sugar all over the place. ENiGMA is meant to run in Node.js, not a web browser. And as such, we already have a known set of ES6 and so on.

**Comment by NuSkooler** [comment]
`PluginModule` and derived classes now use ES6.
